base/server/systemd/SystemdStartupNotifier.java

```diff
@@ -0,0 +1,59 @@
+// --- BEGIN COPYRIGHT BLOCK ---
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; version 2 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// (C) 2020 Red Hat, Inc.
+// All rights reserved.
+// --- END COPYRIGHT BLOCK ---
+
+package com.netscape.cmscore.systemd;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+
+import com.netscape.cmscore.apps.StartupNotifier;
+
+public class SystemdStartupNotifier implements StartupNotifier {
+    interface Systemd extends Library {
+        public int sd_booted();
+        public int sd_notify(int unset_environment, String state);
+    }
+
+    /* loadLibrary is deprecated in jna >= 5; replaced with load()
+     * which avoids the cast.  But RHEL 8 has jna 4.5 so we must put
+     * up with deprecation warnings on Fedora. */
```

Comment:
We need to double-check this on all platforms due to modularity.

@cipherboy ^^

Reply:
`jna` package is in RHEL 8 is not in a module.